File: dsl-style/src/main/java/example/dsl_style/entity/Employee.java
package example.dsl_style.entity;

import example.dsl_style.domain.Age;
import example.dsl_style.domain.Salary;
import java.time.LocalDate;
import java.time.LocalDateTime;
import org.seasar.doma.Column;
import org.seasar.doma.Entity;
import org.seasar.doma.GeneratedValue;
import org.seasar.doma.GenerationType;
import org.seasar.doma.Id;
import org.seasar.doma.Metamodel;
import org.seasar.doma.OriginalStates;
import org.seasar.doma.SequenceGenerator;
import org.seasar.doma.Transient;
import org.seasar.doma.Version;

@Entity(listener = EmployeeListener.class, metamodel = @Metamodel)
public class Employee {

  @Id
  @GeneratedValue(strategy = GenerationType.SEQUENCE)
  @SequenceGenerator(sequence = "EMPLOYEE_SEQ")
  Integer id;

  String name;

  Age age;

  Salary salary;

  @Column(name = "JOB_TYPE")
  JobType jobType;

  LocalDate hiredate;

  @Column(name = "DEPARTMENT_ID")
  Integer departmentId;

  @Version
  @Column(name = "VERSION")
  Integer version;

  LocalDateTime insertTimestamp;

  LocalDateTime updateTimestamp;

  @Transient Department department;

  @OriginalStates Employee originalStates;

  public Integer getId() {
    return id;
  }

  public void setId(Integer id) {
    this.id = id;
  }

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public Age getAge() {
    return age;
  }

  public void setAge(Age age) {
    this.age = age;
  }

  public Salary getSalary() {
    return salary;
  }

  public void setSalary(Salary salary) {
    this.salary = salary;
  }

  public JobType getJobType() {
    return jobType;
  }

  public void setJobType(JobType jobType) {
    this.jobType = jobType;
  }

  public LocalDate getHiredate() {
    return hiredate;
  }

  public void setHiredate(LocalDate hiredate) {
    this.hiredate = hiredate;
  }

  public Integer getDepartmentId() {
    return departmentId;
  }

  public void setDepartmentId(Integer departmentId) {
    this.departmentId = departmentId;
  }

  public LocalDateTime getInsertTimestamp() {
    return insertTimestamp;
  }

  public void setInsertTimestamp(LocalDateTime insertTimestamp) {
    this.insertTimestamp = insertTimestamp;
  }

  public LocalDateTime getUpdateTimestamp() {
    return updateTimestamp;
  }

  public void setUpdateTimestamp(LocalDateTime updateTimestamp) {
    this.updateTimestamp = updateTimestamp;
  }

  public Integer getVersion() {
    return version;
  }

  public void setVersion(Integer version) {
    this.version = version;
  }

  public Department getDepartment() {
    return department;
  }

  public void setDepartment(Department department) {
    this.department = department;
  }

  @Override
  public String toString() {
    return "Employee{"
        + "id="
        + id
        + ", name='"
        + name
        + '\''
        + ", age="
        + age
        + ", salary="
        + salary
        + ", jobType="
        + jobType
        + ", hiredate="
        + hiredate
        + ", departmentId="
        + departmentId
        + ", version="
        + version
        + ", insertTimestamp="
        + insertTimestamp
        + ", updateTimestamp="
        + updateTimestamp
        + ", department="
        + department
        + '}';
  }
}

File: dsl-style-kotlin/src/main/kotlin/example/dsl_style_kotlin/entity/Employee.kt
package example.dsl_style_kotlin.entity

import example.dsl_style_kotlin.domain.Age
import example.dsl_style_kotlin.domain.Salary
import org.seasar.doma.Column
import org.seasar.doma.Entity
import org.seasar.doma.GeneratedValue
import org.seasar.doma.GenerationType
import org.seasar.doma.Id
import org.seasar.doma.Metamodel
import org.seasar.doma.OriginalStates
import org.seasar.doma.SequenceGenerator
import org.seasar.doma.Transient
import org.seasar.doma.Version
import java.time.LocalDate
import java.time.LocalDateTime

@Entity(listener = EmployeeListener::class, metamodel = Metamodel())
class Employee {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    @SequenceGenerator(sequence = "EMPLOYEE_SEQ")
    var id: Int? = null
    var name: String? = null
    var age: Age? = null
    var salary: Salary? = null

    @Column(name = "JOB_TYPE")
    var jobType: JobType? = null
    var hiredate: LocalDate? = null

    @Column(name = "DEPARTMENT_ID")
    var departmentId: Int? = null

    @Version
    @Column(name = "VERSION")
    var version: Int? = null
    var insertTimestamp: LocalDateTime? = null
    var updateTimestamp: LocalDateTime? = null

    @Transient
    var department: Department? = null

    @OriginalStates
    var originalStates: Employee? = null

    override fun toString(): String {
        return (
            "Employee{" +
                "id=" +
                id +
                ", name='" +
                name +
                '\'' +
                ", age=" +
                age +
                ", salary=" +
                salary +
                ", jobType=" +
                jobType +
                ", hiredate=" +
                hiredate +
                ", departmentId=" +
                departmentId +
                ", version=" +
                version +
                ", insertTimestamp=" +
                insertTimestamp +
                ", updateTimestamp=" +
                updateTimestamp +
                ", department=" +
                department +
                '}'
            )
    }
}
